### ISSUE TITLE
Messgerät in Types aufnehmen

### ISSUE THREAD

**Issue by BenAhrdt** [ioBroker/ioBroker.type-detector#53]
Ein Gerät Messgerät sollte mit aufgenommen werden.
Rollen wären zusätzlich
Spannung, Strom, Leistung,
Energy ...

Das Gerät sollte auch nicht nur auf Elektrische Energie beschränkt sein, sondern auch Wasser, Gas, Öl  empfangen können.

**Comment by DutchmanNL** [comment]
Please cross check with https://github.com/ioBroker/ioBroker.docs/pull/423

**Comment by Apollon77** [comment]
I think when that these will be multiple devices. One for energy (aka power/current) and others for other types that also require other roles. 

Issue on hold till sync with matter device types.

**Comment by BenAhrdt** [comment]
Possible roles are for instance:

Power
Voltage
Current
Energy

Maby Power.active, power.reactive

Oder Energy.cosumed, Energy.delivered

**Comment by mcm1957** [comment]
> Possible roles are for instance:
> 
> Power Voltage Current Energy
> 
> Maby Power.active, power.reactive
> 
> Oder Energy.cosumed, Energy.delivered

Problematik could be that those roles are used by i e. Solar inverters, heating, batteries etc too. And I do not think that a solar inverter or a bettery should be detected as measurement device ....

In addition Measuremt has many Variante
Deoending ion the physical United measured. Could ve energy but also temperature,speed,Operation time ...

**Comment by BenAhrdt** [comment]
This issue dont mean the measured value itself, but the device eg. "smart-meter".

**Comment by mcm1957** [comment]
> This issue dont mean the measured value itself, but the device eg. "smart-meter".

YES, but measurement device is not ONE type of devices. A smart-meter is not the same as a lux-meter. So I guess that the term measurement (-device) is to general.

**Comment by BenAhrdt** [comment]
Ok. So using "Info" is better.? 😉

**Comment by Garfonso** [comment]
IMHO we should create a type for every specific sensor/value-role. We have this for some sensors, like flood-alarm-sensor or fire-alarm-sensor. This should be extended to a lot of the value-roles.

**Comment by mcm1957** [comment]
The type-detector tries to detect device-types based on roles from states. Has it ever been discussed to use the device object is an aid? This would improve detecttion for Adapgetrs supportting real devices. (It would not work with general adapters like mqtt of course).

I would suggest to use the existing rile attribuite and to (optionally) add ist to objects of type device (and eventuall channel) and specifiy roles for devices there, i.e. "measuerent", "light", "oven", "climate" etc. with those roles matching matter devices to avoid new confusion. If a device object specifies a device type, detection would be easy.

Obj course detection based in states must be available too for use with general adapers like mqtt etc.

**Comment by mcm1957** [comment]
> This should be extended to a lot of the value-roles.

If done please do not forget to sync level.* too as IMHO every value role should have a matching level role and vice versa.

**Comment by Garfonso** [comment]
> The type-detector tries to detect device-types based on roles from states. Has it ever been discussed to use the device object is an aid? This would improve detecttion for Adapgetrs supportting real devices. (It would not work with general adapters like mqtt of course).

I think this is a bit of the problem with the current situation... type-detector, in the first place, just tries to identify "types", which means some kind of functionality, so that adapters using the library can use this functionality (withour requiring and implemetation or manual setup for every adapter). That is something different than devices. Devices can have multiple functionalities. One could even argue, that the "indicators" like connection-problem, empty battery etc. should not be connected to a temperature sensor but are functionalities on their own. 

The current development seems to go into "devices", which leads to a lot of types "gathering" a lot of optional states, like, for example, the energy measurment in plugs. In the end it makes it harder to use the results, because, either, the optional stuff is ignored or you have to rip the "types" appart anyway and process them individually (which is harder, too, because there is a convetion but no guarantee about the Names, for example ON / ACTUAL are a bit different for a `light` and a `dimmer` type). 

So, IMHO we should not try to make type-detector a "device detector". We should have the device-functionality somewhere else and it probably makes perfect sense to extend the device object with more optional parameters (and have type-detector look at those, too). Like the connection-states, that admin supports (but AFAIK type-detector not yet).
